lib/infrastructure/core/injection.dart

```diff
@@ -6,6 +6,6 @@ import 'injection.config.dart';
 final GetIt getIt = GetIt.instance;
 
 @injectableInit
-void configureInjection() {
-  $initGetIt(getIt);
+Future<void> configureInjection() async {
```

Comment:
I'm not sure this is a good way to do it. You'll have to await the configuration nonetheless? If this is to get sharedpreference as a singleton, I don't think this is a good idea.